crates/driver/src/tests/cases/verify_asset_flow/liquidity.rs

```diff
@@ -0,0 +1,478 @@
+//! Test that the asset flow verification behaves as expected for liquidity
+//! orders. See [`competition::solution::settlement::Verified`].
+
+use {
+    crate::{
+        domain::{
+            competition::{self, auction},
+            eth,
+        },
+        infra,
+        tests::{self, cases::SOLVER_NAME, hex_address, setup},
+    },
+    itertools::Itertools,
+    serde_json::json,
+};
+
+#[derive(Debug)]
+struct LiquidityOrder {
+    sell: eth::Asset,
+    buy: eth::Asset,
+    side: competition::order::Side,
+    executed: eth::U256,
+    partially_fillable: bool,
+}
+
+struct TestCase {
+    valid: bool,
+    liquidity_orders: fn(sell_token: eth::H160, buy_token: eth::H160) -> Vec<LiquidityOrder>,
+}
+
+#[tokio::test]
+#[ignore]
+async fn valid_liquidity_asset_flow() {
+    crate::boundary::initialize_tracing("debug,hyper=warn");
+
+    let test_cases = [
```

Comment:
Would be nice to add a small comment explaining the test setup here. Otherwise you have to read the whole test first to make sense of these cases. AFAICS you set up a single user order where all but 50 units of `sell_token` get used by uniswap which returns all but 50 units of the `buy_token`. Meaning that the liquidity orders may use `50 sell_token` and have to produce `50 buy_token` in total. Is that correct?